### DISCUSSION TITLE
How to use my own regularization images

### DISCUSSION THREAD

**Discussion by SAC020** [TheLastBen/fast-stable-diffusion#543] [Q&A] [answered]
Hi,

Been trying all combinations of parameters (number of steps, % of text training, faces/no faces) to train a BDSM model - and I am sort of failing. 

What I did so far was to provide an image set of a specific woman in the specific BDSM pose, from various angles, captioned woman_specificbdsmpose (1..n), and it sort of works but it's very hard to put the model into specific "locations" or clothing (I suppose this would be called "styling"?)

It occurs to me that I should use a set of regularization images of women in the specific bdsm pose instead? Is this a better approach?

If so:
- where do I put the regularization images?
- how do I caption them?
- how to I specify to the trainer to use my own set, instead of the provided female set?

TYVM!

**Comment by TheLastBen** [comment]
You need to keep the text encoder low to be able to apply style, and don't use existing words in the images names, use random unknown identifiers

**Reply by SAC020** [reply]
Thank you, I understand what you mean and I will test it.

However, if I want to also test a training using my own set of reg images, how do I do it?

**Reply by TheLastBen** [reply]
set contain faces to "Female", than go in the folder in your gdrive :  `"Fast-Dreambooth/Sessions/-your session name-/Regularization_images/Women"` remove all the images inside and upload exactly 200 reg images to that folder

**Reply by SAC020** [reply]
And then I redo the training from the beginning? Because that folder does not exist until I do a training.

If I redo the training, won't the script overwrite my images?

**Comment by SAC020** [comment]
Also, to specify that my main difficulty is not transferring the face of the model to other women, that is working quite fine. My main problem is the BDSM pose, it overlaps with existing vocabulary (e.g. "suspended") but which is implemented poorly in standard SD

**Comment by TheLastBen** [comment, marked as answer]
It looks like you need the old method for this : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/Dreambooth/fast_DreamBooth-Old-Method.ipynb

**Reply by SAC020** [reply]
For the old method, does it matter if I mix pics of various resolutions, as long as they are square? The new method doesn't seem to care about resolution, not sure how the old one behaves

**Reply by TheLastBen** [reply]
it's the same

**Reply by SAC020** [reply]
> It looks like you need the old method for this : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/Dreambooth/fast_DreamBooth-Old-Method.ipynb

All right!! Will need some finetuning with the sample set and the class set, but this is _definitely_ the right approach / the direction what I was looking for!

May I suggest you add the "end session when finished" feature to this method as well, as to not waste computing units?

TYVM indeed!

**Reply by SAC020** [reply]
BTW, I am getting this warning (?):

`The config attributes {'clip_sample': False} were passed to PNDMScheduler, but are not expected and will be ignored. Please verify your scheduler_config.json configuration file.`

I don't know what it means, it does not break the training

**Comment by SAC020** [comment]
when captioning the subject photos for the old model, does the caption need to include the unique token / session name?

**Reply by TheLastBen** [reply]
yes, include the unique identifier to be able to summon it easily in the prompt

**Reply by SAC020** [reply]
Coming back with another question on regularization images: what exactly is the role of the class images? (subject images is clear)

If the subject is "token a woman doing stuff" and the class regularization images contain xxx images of women who are all naked, will this result in a high propensity of naked women doing stuff being generated?

**Reply by TheLastBen** [reply]
regularization images are images of the same class of the instance images, their goal is to reduce the weight of trained subject to avoid overfitting, but personally I find them unnecessary with the new method

**Reply by nawnie** [reply]
did you nuke the contains faces option don't see it on newest repo

**Reply by TheLastBen** [reply]
Yep, it causes more trouble than positive results, so I halted it for now
